### PR TITLE
IPv6 support added

### PR DESCRIPTION
**Pull request by HeMan**
Copied code from stale PR https://github.com/me-no-dev/AsyncTCP/pull/105

### CODE REVIEW

**Comment by HeMan** [comment]
This should solve https://github.com/esphome/issues/issues/5095

**Review by jesserockz — APPROVED** [review]

**Comment by mathieucarbou** [comment]
Hello,
AsyncTCP is not compiling on ESP32 following the merge and publication of the new 2.1.0 tag:
```
.pio/libdeps/oss-esp32/AsyncTCP-esphome/src/AsyncTCP.cpp:753:2: error: #else without #if
 #else
  ^~~~
.pio/libdeps/oss-esp32/AsyncTCP-esphome/src/AsyncTCP.cpp:755:2: error: #endif without #if
 #endif
  ^~~~~
.pio/libdeps/oss-esp32/AsyncTCP-esphome/src/AsyncTCP.cpp:1020: error: unterminated #if
 #if LWIP_IPV4 && LWIP_IPV6
```

File: src/AsyncTCP.cpp [hunk 1]
```diff
@@ -724,22 +716,39 @@ bool AsyncClient::connect(IPAddress ip, uint16_t port){
...
+
     if(!_start_async_task()){
       log_e("failed to start task");
       return false;
     }
-    
+
     err_t err = dns_gethostbyname(host, &addr, (dns_found_callback)&_tcp_dns_found, this);
     if(err == ERR_OK) {
-#if LWIP_IPV4 && LWIP_IPV6
```
Comment (ForceConstant) [inline src/AsyncTCP.cpp:742]: I think the removal of this ifdef might be causing the build issues.

Comment (mathieucarbou) [reply src/AsyncTCP.cpp:742]: That's what I saw too. I switched back to the previous version in my project.

**Pull request merged**